editor/src/clj/editor/pipeline.clj

```diff
@@ -151,21 +151,25 @@
       (and (.exists f) (= mtime (.lastModified f)) (= size (.length f))))))
 
 (defn- to-disk! [artifact content-hash]
-  (assert (some? (:content artifact)))
+  (assert (or (some? (:build-fn artifact)) (some? (:content artifact))))
```

Comment:
I love this change!

Although I'm not entirely happy about calling this `:build-fn`, since we also have `:build-fn` in the `build-targets` that means something else entirely.

Maybe something like `:write-content-fn` would be better?